File: backend/test_connection.py
# test_connection.py
import pyodbc
import sys

def test_connection():
    # Configuración de la conexión
    server = 'MEZADESKTOP\SQLEXPRESS'  # Ej: 'localhost', '192.168.1.100', 'nombre_instancia'
    database = 'ClientesDB'
    username = 'sa'
    password = 'mezasql'
    
    try:
        # Cadena de conexión
        connection_string = f'DRIVER={{ODBC Driver 17 for SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}'
        
        # Intentar conexión
        print("Intentando conectar a SQL Server...")
        conn = pyodbc.connect(connection_string)
        
        # Crear cursor y ejecutar consulta simple
        cursor = conn.cursor()
        cursor.execute("SELECT @@VERSION")
        version = cursor.fetchone()
        
        print("✅ Conexión exitosa!")
        print(f"Versión de SQL Server: {version[0]}")
        
        # Cerrar conexión
        cursor.close()
        conn.close()
        return True
        
    except pyodbc.Error as e:
        print(f"❌ Error de conexión: {e}")
        return False
    except Exception as e:
        print(f"❌ Error inesperado: {e}")
        return False

if __name__ == "__main__":
    test_connection()

File: backend/test_connection.js
// test-connection.js
// test_connection.js - PRUEBA ESTAS OPCIONES
const sql = require('mssql');

// Probar estas configuraciones UNA POR UNA:
const configOptions = [
    {
        server: 'localhost\\SQLEXPRESS',
        database: 'master',
        user: 'sa',
        password: 'mezasql',
        options: { encrypt: false, trustServerCertificate: true }
    },
    {
        server: '.\\SQLEXPRESS',
        database: 'master',
        user: 'sa',
        password: 'mezasql',
        options: { encrypt: false, trustServerCertificate: true }
    },
    {
        server: '127.0.0.1\\SQLEXPRESS',
        database: 'master',
        user: 'sa',
        password: 'mezasql',
        options: { encrypt: false, trustServerCertificate: true }
    }
];

async function testConnection(config) {
    try {
        console.log(`🔌 Probando: ${config.server}...`);
        await sql.connect(config);
        console.log('✅ CONEXIÓN EXITOSA!');
        const result = await sql.query`SELECT @@VERSION as version`;
        console.log('Versión:', result.recordset[0].version);
        await sql.close();
        return true;
    } catch (err) {
        console.log('❌ Error:', err.message);
        return false;
    }
}

async function testAll() {
    for (let i = 0; i < configOptions.length; i++) {
        console.log(`\n--- Opción ${i + 1} ---`);
        const success = await testConnection(configOptions[i]);
        if (success) {
            console.log(`\n🎉 CONFIGURACIÓN GANADORA:`);
            console.log(JSON.stringify(configOptions[i], null, 2));
            return;
        }
    }
    console.log('\n😞 Ninguna funcionó. Probemos algo más...');
}

testAll();